### ISSUE TITLE
tokio crate doesn't compile on FreeBSD

### ISSUE THREAD

**Issue by Follpvosten** [tokio-rs/tokio#219]
Hello,
i am currently developing a telegram bot with telebot, which will be running on a FreeBSD server.
When i try compiling my project with cargo, i get the following error:

```
   Compiling tokio v0.1.3
error[E0425]: cannot find function 'hup' in this scope
   --> /home/jonas/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.3/src/reactor/poll_evented.rs:443:9
    |
443 |         hup() | UnixReady::aio() | UnixReady::lio()
    |         ^^^ not found in this scope

error: aborting due to previous error

If you want more information on this error, try using "rustc --explain E0425"
error: Could not compile 'tokio'.
```

As a dirty workaround, i replaced the erroring line with:
`Ready::from_usize(UnixReady::hup().as_usize()) | UnixReady::aio() | UnixReady::lio()`
Now it compiles, but that's not a solution.
`hup()` should probably be defined for FreeBSD. Can i help with that? What should it do?

To reproduce, install the current nightly toolchain using rustup on a FreeBSD system, clone my [test project](https://github.com/Follpvosten/test_telegram_bot), and run `cargo build` for it.

**Comment by kpp** [comment]
https://github.com/tokio-rs/tokio/issues/158 again

**Comment by carllerche** [comment]
FreeBSD needs to be added to CI to avoid these errors recurring.

**Comment by rockstar** [comment]
FreeBSD support in tokio is about to become mission critical for my application. I'm happy to support some CI resources to prevent this from happening in the future. Please let me know how I can help.

**Comment by carllerche** [comment]
Good question.

The quick fix is this: #228, which compiles, but does not run.

The better fix would be to get some hardware running the various operating systems, and then figure out a way to have travis request the hardware to run a build on the hardware.

**Comment by carllerche** [comment]
I'm going to close this as #248 has merged, but ideally we would figure out how to get some real hardware to run the tests on various OSes.
